### PR TITLE
New Query, ImageQuery, and Cookie helper classes; Updated README to explain capabilities

### PR DESCRIPTION
**Pull request by PFython**
This helper class is a simple wrapper around the existing code to provide a quick start for users of the library and make all config, attributes, and output available using convenient '.attribute' notation.

See latest draft [README](https://github.com/PFython/EdgeGPT/tree/patch-1) for explanation of new classes/features.

### CODE REVIEW

**Review by et2300 — CHANGES_REQUESTED** [review]
ok

File: src/EdgeGPT.py
Comment (VMTask) [inline src/EdgeGPT.py:1]: Remove the demo function.

Comment (PFython) [reply src/EdgeGPT.py:1]: Yep, just included to illustrate usage.  If this Pull Request is approved, I'm happy to submit an updated README.

**Review by acheong08 — COMMENTED** [review]
What is this for? I can't think of how this could be used

**Comment by PFython** [comment]
Hi Acheong... sorry it it wasn't obvious.

The point of offering this is that it can be used, as I use it, to save time and avoid the need for everyone who uses your this repo to create their own boilerplate code.

-If you want to submit a simple query with default values, it does that as simply as ```q = Query(<prompt>)```
- It provides "batteries included" parsing of the response, including just extracting any Python snippets, just extracting the sources (optionally as a dictionary), just extracting the suggestions, or just extracting the body of the output.
- It automatically keeps a running list of queries you've made in .index for future use.  Individual attributes record things like what your prompt was and what style you asked for.

Surely everyone who uses your code will want to do these things routinely?  And since every request also needs cookies to be preloaded, why not include this in the 'core' of your solution, rather than leaving it for each of your 7,000 fans to write slightly different versions of essentially the same thing?

I've suggested it as a separate class that can be simply imported with ```from EdgeGPT import Query``` so that those who want to can still interface with your code as currently set out in your README; This class just abstracts away some super-common processes, and also some unnecessary complications like the wss link (which is apparently unnecessary anyway, and you don't explain what other options people might use for different geographies).  It also provides sensible default values so that if you wish you can just supply a prompt and nothing more.

Alternatively you could merge this with your original Chatbot class to achieve the same thing - convenient "batteries included" shortcut, default values, memorable and easy to access class and instance attributes, and methods for the things that surely everybody will want - like to parse the main output.

I'm happy to work some more on this if you like the features but want it implemented in a particular way.  If not then no stress - I'm already benefiting from it, thanks to your efforts and skill.

PS I've also found written a little routine to cycle through multiple cookies files, hence overcoming the 200 requests per user per day restriction. 
-
```

**Comment by acheong08** [comment]
Thanks for the detailed explanation. Can you add a tiny bit of docs about the `.attribute` notation? I will merge this since it is pretty helpful.

File: src/EdgeGPT.py [hunk 1]
```diff
@@ -874,5 +874,108 @@ def main() -> None:
...
+            else:
+                continue
+        return sources_dict
+
+    @property
+    def python(self):
+        """Extract and join any snippets of Python code in the response"""
+        codeblocks = self.output.split("```python\n")
+        codeblocks = [x.split("```")[0] for x in codeblocks if "```" in x]
+        return "\n".join(codeblocks)
```
Comment (acheong08) [inline src/EdgeGPT.py:952]: Can this be language independent?

Perhaps split by ``` and take the first word as a language property and remove \n

Thanks!

**Review by acheong08 — CHANGES_REQUESTED** [review]

File: src/EdgeGPT.py (hunk 1 above)
Comment (PFython) [reply src/EdgeGPT.py:952]: My pleasure and good suggestion regarding multiple languages.  Working on it today.

**Comment by PFython** [comment]
OK changes made to EdgeGPT.py and README.md to explain the new helper classes.

Hope you like it :)

**Review by acheong08 — DISMISSED** [review]

**Review by acheong08 — APPROVED** [review]

**Pull request merged**